### PR TITLE
ENG-12584

### PR DESCRIPTION
**Pull request by sycai**
Enable JDBCLoader to be topology-aware

### CODE REVIEW

File: src/frontend/org/voltdb/utils/JDBCLoader.java [hunk 1]
```diff
@@ -459,10 +459,21 @@ private static void configuration() {
...
 
         for (String server : servers) {
-            client.createConnection(server.trim(), port);
+         // Try connecting servers one by one until we have a success
+            try {
+                client.createConnection(server.trim(), port);
+                break;
+            } catch(IOException e) {
+                // Only swallow the exceptions from Java network or connection problems
+                // Unresolved hostname exceptions will be thrown
```
Comment (brians127) [inline src/frontend/org/voltdb/utils/JDBCLoader.java]: Why did you choose this error handling strategy?

File: src/frontend/org/voltdb/utils/JDBCLoader.java [hunk 2]
```diff
@@ -459,10 +459,21 @@ private static void configuration() {
...
+        config.setTopologyChangeAware(true);
         final Client client = ClientFactory.createClient(config);
 
         for (String server : servers) {
-            client.createConnection(server.trim(), port);
+         // Try connecting servers one by one until we have a success
+            try {
+                client.createConnection(server.trim(), port);
+                break;
+            } catch(IOException e) {
```
Comment (brians127) [inline src/frontend/org/voltdb/utils/JDBCLoader.java]: Not blocker: please put a space after "catch"

File: src/frontend/org/voltdb/utils/JDBCLoader.java (hunk 1 above)
Comment (sycai) [inline src/frontend/org/voltdb/utils/JDBCLoader.java]: An unhandled exception would stop the client from connecting more servers, which is not we want for a topology-aware loader.

File: src/frontend/org/voltdb/utils/JDBCLoader.java [hunk 3]
```diff
@@ -459,10 +459,22 @@ private static void configuration() {
...
+            try {
+                client.createConnection(server.trim(), port);
+                break;
+            } catch (IOException e) {
+                // Only swallow the exceptions from Java network or connection problems
+                // Unresolved hostname exceptions will be thrown
+            }
+        }
+        if (client.getConnectedHostList().isEmpty()) {
+            client.close();
```
Comment (nshi) [inline src/frontend/org/voltdb/utils/JDBCLoader.java]: Catch any exception thrown from this.

File: src/frontend/org/voltdb/utils/JDBCLoader.java [hunk 4]
```diff
@@ -459,10 +459,22 @@ private static void configuration() {
...
             int port) throws Exception {
+        config.setTopologyChangeAware(true);
         final Client client = ClientFactory.createClient(config);
 
         for (String server : servers) {
-            client.createConnection(server.trim(), port);
+         // Try connecting servers one by one until we have a success
+            try {
+                client.createConnection(server.trim(), port);
+                break;
```
Comment (nshi) [inline src/frontend/org/voltdb/utils/JDBCLoader.java]: To leverage client affinity, you should continue to try to connect to all servers.

**Pull request merged**